vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -981,6 +978,109 @@ private void attachComponentRenderer() {
         setDimension(ElementConstants.STYLE_MAX_HEIGHT, maxHeight);
     }
 
+    private void updateOverlayClassName() {
+        getElement().setProperty("overlayClass", getClassName());
+    }
+
+    /**
+     * Adds a CSS class name to the dialog overlay element.
+     *
+     * @param className
+     *            the CSS class name to add, not <code>null</code>
+     */
+    @Override
+    public void addClassName(String className) {
+        HasStyle.super.addClassName(className);
+
+        updateOverlayClassName();
```

Comment:
Probably redundant to call these now when the backing collection already does the same

Reply:
Removed all unnecessary overrides.